### PR TITLE
Dance redux in TypeScript/redux-toolkit

### PR DESCRIPTION
**Pull request by bencodeorg**
Move Dance Party redux store to redux-toolkit / typescript.

No functional changes -- one cleanup was to rename the store to `dance` from `songs`, as it contains some state about the run button (in addition to information about songs).

## Links

- jira ticket: [LABS-54](https://codedotorg.atlassian.net/browse/LABS-54)

## Testing story

Tested manually in a standalone Dance Party project that:
- song metadata loaded successfully into the dropdown of song options
- could swap between songs
- could hit the run button and see my animation

### CODE REVIEW

File: apps/src/dance/danceRedux.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import {createSlice, PayloadAction} from '@reduxjs/toolkit';
+import {registerReducers} from '@cdo/apps/redux';
+
+type SongData = {
```
Comment (sanchitmalhotra126) [inline apps/src/dance/danceRedux.ts]: Nice! Could we move this into a `types.ts` file? I'm guessing we'll want to reference this elsewhere eventually

**Review by sanchitmalhotra126 — APPROVED** [review]
Looks great!

File: apps/src/dance/danceRedux.ts (hunk 1 above)
Comment (fisher-alice) [reply apps/src/dance/danceRedux.ts]: Love TS! Was wondering where these came from and see these keys in `songs.js`. Appreciate that they're declared as a type.

Comment (bencodeorg) [reply apps/src/dance/danceRedux.ts]: Added!

**Review by fisher-alice — APPROVED** [review]
Nice job!

**Pull request merged**